include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_reduce.h

```diff
@@ -66,22 +66,25 @@ __work_group_reduce_kernel(const _NDItemId __item_id, const _Size __n, const _Si
                            const bool __is_full, _TransformPattern __transform_pattern, _ReducePattern __reduce_pattern,
                            _InitType __init, const _AccLocal& __local_mem, _Res* __res_ptr, const _Acc&... __acc)
 {
-    auto __local_idx = __item_id.get_local_id(0);
-    const _Size __group_size = __item_id.get_local_range().size();
     oneapi::dpl::__internal::__lazy_ctor_storage<_Tp> __result;
-    // 1. Initialization (transform part). Fill local memory
-    __transform_pattern(__item_id, __n, __iters_per_work_item, /*global_offset*/ (_Size)0, __is_full,
-                        /*__n_groups*/ (_Size)1, __result, __acc...);
 
-    const _Size __n_items = __transform_pattern.output_size(__n, __group_size, __iters_per_work_item);
-    // 2. Reduce within work group using local memory
-    __result.__v = __reduce_pattern(__item_id, __n_items, __result.__v, __local_mem);
-    if (__local_idx == 0)
+    // 1. Initialization (transform part). Fill local memory
+    if (__transform_pattern(__item_id, __n, __iters_per_work_item, /*global_offset*/ (_Size)0, __is_full,
```

Comment:
Is it possible for this to return true for some items in a work-group and false for others? If so, this may result in hangs as we use group barriers within this conditional.